common/static/common/js/discussion/utils.js

```diff
@@ -421,15 +447,16 @@
             );
         };
 
-        DiscussionUtil.stripLatexHighlight = function(text) {
-            return this.processEachMathAndCode(text, this.stripHighlight);
+        DiscussionUtil.stripLatexHighlight = function(safeHtml) {
+            return this.processEachMathAndCode(safeHtml, this.stripHighlight);
         };
 
         DiscussionUtil.markdownWithHighlight = function(text) {
             var converter;
             text = text.replace(/^\&gt\;/gm, ">");
             converter = Markdown.getMathCompatibleConverter();
-            text = this.unescapeHighlightTag(this.stripLatexHighlight(converter.makeHtml(text)));
+            var safeHtml = converter.makeHtml(text);
```

Comment:
For example, if you know that `converter.makeHtml(text)` returns safe html, then you could use:
```
var safeHtml = HtmlUtils.HTML(converter.makeHtml(text));
```
In this way, safeHtml would be turned into an HTML Snippet in the context in which you can see that is safe to do, and then it can be used with other HtmlUtils functions from this point out (or using toString() where and when you must).